### PR TITLE
File input extended to support state

### PR DESCRIPTION
**Pull request by the-greenman**
This PR updates the file input block to be state aware, and adds "EnabledGetter" functionality to allow the button to be revealed or hidden based on state.

### CODE REVIEW

File: cypress/fixtures/develModeBlocks.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
```
Comment (the-greenman) [inline cypress/fixtures/develModeBlocks.json:1]: I've created this fixture to simplify future testing of these features. 
It defines a standard set of blocks that should be state aware, and removes the need to define blocks in each test. 


Older tests have not yet been refactored to use it.

File: src/app/blocks/file-input-block/file-input-block.component.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import {Component} from '@angular/core';
```
Comment (the-greenman) [inline src/app/blocks/file-input-block/file-input-block.component.ts:1]: There are a fair number of changes required to support this functionality. 
I have been considering adding it to the base class, or creating a new extension of the base class to make it easier to implement. 
Just not this time.

File: cypress/fixtures/develModeBlocks.json (hunk 1 above)
Comment (lukestanley) [reply cypress/fixtures/develModeBlocks.json:1]: Couldn't this sort of comment could go in the JSON itself, and maybe the filename could be more obvious?

Comment (the-greenman) [reply cypress/fixtures/develModeBlocks.json]: Done. Comment and file rename

File: cypress/integration/context_and_state.ts [hunk 3]
```diff
@@ -133,5 +133,14 @@ it('should allow the disabling of the debug block', () => {
...
-
-});
+it('should allow the disabling of the file input block ', () => {
+  cy.intercept('GET', 'https://app.kendra.io/api/TESTING/develmodeblocks', {
+    fixture: 'develModeBlocks.json' }
+    ).as('workflow.json');
+    cy.visit('/TESTING/develmodeblocks');
+    cy.contains('Devel Import').should('not.exist');
+    cy.contains("Enable Devel Mode").click();
+    cy.contains('Devel Import').should('exist');
```
[the remark this answers is not in the document]
Comment (the-greenman) [reply cypress/integration/context_and_state.ts:144]: Yeah, good point. I've changed the label on the button to be more verbose

[the remark this answers is not in the document]
Comment (the-greenman) [reply cypress/integration/context_and_state.ts]: I've updated the labels here

**Review by lukestanley — APPROVED** [review]

**Pull request merged**